steward/src/cork.rs

```diff
@@ -93,6 +99,90 @@ impl steward::contract_call_server::ContractCall for CorkHandler {
+
+        let wallet: LocalWallet = Wallet::from(key);
+
+        let eth_host = config.ethereum.rpc.clone();
+        let provider = Provider::<Http>::try_from(eth_host.clone())
+            .unwrap_or_else(|_| panic!("could not get provider from eth_host {}", eth_host));
+        let chain_id = provider
+            .get_chainid()
+            .await
+            .expect("Could not retrieve chain ID");
+
+        let chain_id =
+            downcast_to_u64(chain_id).expect("Chain ID overflowed when downcasting to u64");
+        let client = SignerMiddleware::new(provider, wallet.clone().with_chain_id(chain_id));
+
+        let cellar_address = request.cellar_id.clone();
+
+        if let Err(err) = cellars::validate_cellar_id(&cellar_address) {
+            return Err(tonic::Status::invalid_argument(err));
+        }
+
+        let address: H160 = cellar_address
+            .parse()
+            .expect("could not parse cellar address");
+
+        let contract_call_data = match request.call_data.clone() {
+            Some(call) => call,
+            None => return Err(tonic::Status::invalid_argument("err")),
+        };
+        let encoded_call = get_encoded_call(contract_call_data).unwrap();
+
+        // (ugly way to write the ABI inline, you can otherwise read it from a file)
+        // let abi: abi::Abi = serde_json::from_str(r#"[{"inputs":[{"internalType":"string","name":"value","type":"string"}],"stateMutability":"nonpayable","type":"constructor"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"address","name":"author","type":"address"},{"indexed":true,"internalType":"address","name":"oldAuthor","type":"address"},{"indexed":false,"internalType":"string","name":"oldValue","type":"string"},{"indexed":false,"internalType":"string","name":"newValue","type":"string"}],"name":"ValueChanged","type":"event"},{"inputs":[],"name":"getValue","outputs":[{"internalType":"string","name":"","type":"string"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"lastSender","outputs":[{"internalType":"address","name":"","type":"address"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"string","name":"value","type":"string"}],"name":"setValue","outputs":[],"stateMutability":"nonpayable","type":"function"}]"#).unwrap();
+
+        let abi = abi::decode(&[abi_type], &encoded_call).unwrap();
+
+        for abi in abi {
+            let abi = abi.to_string();
+
+            let abi: abi::Abi = serde_json::from_str(&abi).unwrap();
+
+            // create the contract object at the address
+            let contract = Contract::new(address, abi, &client);
+
+            // Calling constant methods is done by calling `call()` on the method builder.
+            // (if the function takes no arguments, then you must use `()` as the argument)
+            contract
+                .method::<_, String>("getValue", ())
+                .unwrap()
+                .call()
```

Comment:
`call` works for non-mutating state contract calls. In order to broadcast a transaction you would have to use `send`, which has some more steps involved. Here's an example from the gravity bridge, though it's wrapping it in a timeout:

https://github.com/PeggyJV/gravity-bridge/blob/fb045a1867b723123fdc54d01f8e8be936a7189e/orchestrator/ethereum_gravity/src/logic_call.rs#L64-L90